docs/generator.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/eris-ltd/common/go/docs"
+	commands "github.com/eris-ltd/eris-db/cmd"
+
+	"github.com/eris-ltd/eris-db/version"
+	"github.com/spf13/cobra"
+)
+
+// Repository maintainers should customize the next two lines.
+var Description = "Blockchain Client"                                         // should match the docs site name
+var RenderDir = fmt.Sprintf("./docs/documentation/db/%s/", version.VERSION) // should be the "shortversion..."
+
+// The below variables should be updated only if necessary.
+var Specs = []*docs.Entry{}
+var Examples = []*docs.Entry{}
+var SpecsDir = "./docs/specs"
+var ExamplesDir = "./docs/examples"
+
+type Cmd struct {
+	Command     *cobra.Command
+	Entry       *docs.Entry
+	Description string
+}
+
+func RenderFiles(cmdRaw *cobra.Command, tmpl *template.Template) error {
+	this_entry := &docs.Entry{
+		Title:          cmdRaw.CommandPath(),
+		Specifications: Specs,
+		Examples:       Examples,
+		BaseURL:        strings.Replace(RenderDir, ".", "", 1),
+		Template:       tmpl,
+		FileName:       docs.GenerateFileName(RenderDir, cmdRaw.CommandPath()),
+	}
+
+	cmd := &Cmd{
+		Command:     cmdRaw,
+		Entry:       this_entry,
+		Description: Description,
+	}
+
+	for _, command := range cmd.Command.Commands() {
+		RenderFiles(command, tmpl)
+	}
+
+	if !cmd.Command.HasParent() {
+		entries := append(cmd.Entry.Specifications, cmd.Entry.Examples...)
+		for _, entry := range entries {
+			entry.Specifications = cmd.Entry.Specifications
+			entry.Examples = cmd.Entry.Examples
+			entry.CmdEntryPoint = cmd.Entry.Title
+			entry.BaseURL = cmd.Entry.BaseURL
+			if err := docs.RenderEntry(entry); err != nil {
+				return err
+			}
+		}
+	}
+
+	outFile, err := os.Create(cmd.Entry.FileName)
+	if err != nil {
+		return err
+	}
+	defer outFile.Close()
+
+	err = cmd.Entry.Template.Execute(outFile, cmd)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func main() {
+	// Repository maintainers should populate the top level command object.
+	// pm := commands.EPMCmd
```

Comment:
The template -> https://github.com/eris-ltd/common/blob/master/go/docs/docs_generator.go#L75-L123 is rendered based on cobra's Command struct.